src/code/audio_heap.c

```diff
@@ -1,106 +1,1386 @@
+            reverb = reverb;
+            for (j = 0; j < gAudioContext.gAudioBufferParameters.updatesPerFrame; j++) {
+                mem = func_800DE258(&gAudioContext.gNotesAndBuffersPool, 0x340);
+                reverb->items[0][j].toDownsampleLeft = mem;
+                reverb->items[0][j].toDownsampleRight = mem + 0x1A0 / sizeof(s16);
+                mem = func_800DE258(&gAudioContext.gNotesAndBuffersPool, 0x340);
+                reverb->items[1][j].toDownsampleLeft = mem;
+                reverb->items[1][j].toDownsampleRight = mem + 0x1A0 / sizeof(s16);
+            }
+        }
+
+        if (settings->unk_14 != 0) {
+            reverb->unk_278 = Audio_AllocDmaMemoryZeroed(&gAudioContext.gNotesAndBuffersPool, 0x40);
+            reverb->unk_270 = Audio_AllocDmaMemory(&gAudioContext.gNotesAndBuffersPool, 8 * sizeof(s16));
+            func_800DF5DC(reverb->unk_270, settings->unk_14);
+        } else {
+            reverb->unk_270 = NULL;
+        }
+
+        if (settings->unk_16 != 0) {
+            reverb->unk_27C = Audio_AllocDmaMemoryZeroed(&gAudioContext.gNotesAndBuffersPool, 0x40);
+            reverb->unk_274 = Audio_AllocDmaMemory(&gAudioContext.gNotesAndBuffersPool, 8 * sizeof(s16));
+            func_800DF5DC(reverb->unk_274, settings->unk_16);
+        } else {
+            reverb->unk_274 = NULL;
+        }
+    }
+
+    Audio_InitSequencePlayers();
+    for (i = 0; i < gAudioContext.gAudioBufferParameters.numSequencePlayers; i++) {
+        func_800EC734(i);
+        Audio_ResetSequencePlayer(&gAudioContext.gSequencePlayers[i]);
+    }
+
+    func_800E0634(preset->unk_30, preset->unk_34);
+    func_800E1618(gAudioContext.gMaxSimultaneousNotes);
+    gAudioContext.unk_176C = 0;
+    func_800E3400();
+    func_800E4FB0();
+    func_800E3A14();
+    gAudioContext.unk_4 = 0x1000;
+    func_800E4D94();
+    intMask = osSetIntMask(1);
+    osWritebackDCacheAll();
+    osSetIntMask(intMask);
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/audio_heap/func_800DFBF8.s")
+#endif
+
+void* func_800E04E8(s32 poolIdx, s32 id) {
+    s32 i;
+
+    for (i = 0; i < gAudioContext.unk_2D50.unused; i++) {
+        if (gAudioContext.unk_2D60[i].poolIndex == poolIdx && gAudioContext.unk_2D60[i].id == id) {
+            return gAudioContext.unk_2D60[i].ptr;
+        }
+    }
+    return NULL;
+}
+
+void* func_800E0540(s32 poolIdx, s32 id, u32 size) {
+    void* ret;
+    s32 sp18;
+
+    sp18 = gAudioContext.unk_2D50.unused;
+
+    ret = Audio_Alloc(&gAudioContext.unk_2D50, size);
+    gAudioContext.unk_2D60[sp18].ptr = ret;
+    if (ret == NULL) {
+        return NULL;
+    }
+    gAudioContext.unk_2D60[sp18].poolIndex = poolIdx;
+    gAudioContext.unk_2D60[sp18].id = id;
+    gAudioContext.unk_2D60[sp18].size = size;
+    // @bug missing return? "ret" is in v0, but doing an explicit return
+    // uses an additional register.
+#ifdef AVOID_UB
+    return ret;
+#endif
```

Comment:
We don't have any notion of AVOID_UB in oot (for now at least) so this should probably be omitted?